data/basics/teal.go

```diff
@@ -77,24 +78,7 @@ type StateDelta map[string]ValueDelta
 // equality because an empty map will encode/decode as nil. So if our generated
 // map is empty but not nil, we want to equal a decoded nil off the wire.
 func (sd StateDelta) Equal(o StateDelta) bool {
-	// Lengths should be the same
-	if len(sd) != len(o) {
-		return false
-	}
-	// All keys and deltas should be the same
-	for k, v := range sd {
-		// Other StateDelta must contain key
-		ov, ok := o[k]
-		if !ok {
-			return false
-		}
-
-		// Other StateDelta must have same value for key
-		if ov != v {
-			return false
-		}
-	}
-	return true
+	return maps.Equal(sd, o)
```

Comment:
Comment goes into a lot of detail about the `nil` checking choice, but that is exactly the same choiie that maps.Equal made.  Unit tests already exist to confirm.